src/content/docs/en/basics/layouts.mdx

```diff
@@ -258,6 +258,52 @@ const { title } = Astro.props.frontmatter || Astro.props;
 </html>
 ```
 
+### Working with TypeScript
+
+Any astro layout can be modified to introduce typesafety & autocompletion by providing the types for your props:
+
+```astro ins={2-7} title="src/components/MyLayout.astro"
+---
+interface Props { 
+	title: string;
+  description: string;
+  publishDate: Date;
+  viewCount: number;
+}
+const { title, description, publishDate, viewCount } = Astro.props;
+---
+<html lang="en">
+  <head>
+    <meta charset="UTF-8">
+    <meta name="description" content={description}>
+    <title>{title}</title>
+  </head>
+  <body>
+    <header>
+      <p>Published on {publishDate}</p>
+      <p>Viewed by {viewCount} folks</p>
+    </header>
+    <main>
+      <slot />
+    </main>
+  </body>
+</html>
+```
+
+You can override Astro's default `Props` to handle situations where your layout recieves props of different types:
```

Comment:
```suggestion
You can also create custom `Props` to handle situations where your layout receives props of different types:
```
@at-the-vr , what do you think if we just go a little more general like this? :sweat_smile:

Reply:
Apologies for being blunt but the whole sentence `You can override Astro's ... props of different types:` happens to be incorrect with what I know of Types and Astro handling them. The suggestion is not doing any good here honestly.

Props can be custom in Astro and that's correct, but `Astro.props.frontmatter` is very specific. I believe its mentioned in [Markdown Layout Props](<https://docs.astro.build/en/basics/layouts/#markdown-layout-props>) - Please correct me if I happen to be faulty here 😓 

So when someone attempts the following (similar to OP's Issue linked)
```js
---
// snippet from Blog Template
type Props = CollectionEntry<"blog">["data"];

const { title, description, pubDate, updatedDate, heroImage } = Astro.props.frontmatter || Astro.props;
---
```
They receive the Property `frontmatter` does not exists Error.

With respect to the Blog template this issue can be resolved by CustomProps like this:
```diff
+ type CustomProps = CollectionEntry<"blog">["data"] & {
  title: string,
  description: string,
  pubDate: Date,
  updatedDate: Date,
  heroImage: string,
}

+ const { title, description, pubDate, updatedDate, heroImage } :CustomProps = 
 Astro.props.frontmatter || Astro.props;
---
```
and that's what I want to address in that sentence

My direction: Here's a way you can avoid Property `frontmatter` does not exists error
Docs direction: Have to rely on Sarah and Team Astro for this one 🙏